From 868f20242ed99710a9ff9d17d12cbb124619920f Mon Sep 17 00:00:00 2001
From: Ahmad Saleem <ahmad.saleem792@gmail.com>
Date: Tue, 21 May 2024 00:01:18 -0700
Subject: [PATCH] `m_maxMessageSize` should be `infinity()` instead of `max()`
 in `RTCSctpTransport.h`

https://bugs.webkit.org/show_bug.cgi?id=274025
rdar://problem/128306030

Reviewed by Youenn Fablet.

This patch aligns WebKit with web specification [1]:

[1] https://w3c.github.io/webrtc-pc/#dfn-update-the-data-max-message-size

"If both remoteMaxMessageSize and canSendSize are 0, set
[[MaxMessageSize]] to the positive Infinity value."

* Source/WebCore/Modules/mediastream/RTCSctpTransport.h:
(double m_maxMessageSize):
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCSctpTransport-maxMessageSize-expected.txt: Rebaselined

Canonical link: https://commits.webkit.org/279039@main
---
 .../webrtc/RTCSctpTransport-maxMessageSize-expected.txt   | 8 ++++----
 Source/WebCore/Modules/mediastream/RTCSctpTransport.h     | 5 +++--
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/Source/WebCore/Modules/mediastream/RTCSctpTransport.h b/Source/WebCore/Modules/mediastream/RTCSctpTransport.h
index d8bf08666adb..c4f1e85bef58 100644
--- a/Source/WebCore/Modules/mediastream/RTCSctpTransport.h
+++ b/Source/WebCore/Modules/mediastream/RTCSctpTransport.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021 Apple Inc. All rights reserved.
+ * Copyright (C) 2021-2024 Apple Inc. All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions
@@ -73,7 +73,8 @@ private:
     UniqueRef<RTCSctpTransportBackend> m_backend;
     Ref<RTCDtlsTransport> m_transport;
     RTCSctpTransportState m_state { RTCSctpTransportState::Connecting };
-    double m_maxMessageSize { std::numeric_limits<double>::max() };
+    // https://w3c.github.io/webrtc-pc/#dfn-update-the-data-max-message-size
+    double m_maxMessageSize { std::numeric_limits<double>::infinity() };
     std::optional<unsigned short> m_maxChannels;
 };
 
-- 
2.45.1

